x-pack/libbeat/management/generate_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package management
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func TestBareConfig(t *testing.T) {
+	// config with datastreams, metadata, etc, removed
+	rawExpected := proto.UnitExpectedConfig{
```

Comment:
I'm not sure this is an exhaustive test. You would need message variants where each level of nested message is individually nil. For example with `stream.data_stream.source` a test that sets `stream` to `nil` won't catch missing checks for `source` being nil because you'll just short circuit the code block.

Short of writing exhaustive tests like this the best way to guard against this is to never directly access the message fields, use the generated `GetX` accessors chained together. See https://github.com/elastic/beats/pull/32673/files#r960123520